apis/keda/v1alpha1/scaledobject_types.go

```
@@ -56,6 +56,10 @@ const ScaledObjectTransferHpaOwnershipAnnotation = "scaledobject.keda.sh/transfe
 const ValidationsHpaOwnershipAnnotation = "validations.keda.sh/hpa-ownership"
 const PausedReplicasAnnotation = "autoscaling.keda.sh/paused-replicas"
 const PausedAnnotation = "autoscaling.keda.sh/paused"
+const FallbackBehaviorStatic = "static"
+const FallbackBehaviorCurrentReplicas = "currentReplicas"
```

This is the current behavior without fallback, does it make sense adding it also as part of the fallback? I mean, not configured fallback feature will just keep the current replicas

---

I added this one because of this request of zroubalik: https://github.com/kedacore/keda/pull/6464#pullrequestreview-2598459620

wdyt @zroubalik , remove it? Or is there a useful reason to let it stay as is?

---

I belive that for some scalers the workload is scaled to minReplicas if there's a problem in metrics delivery (it reports no activity). Or am I mistaken?

---

> I belive that for some scalers the workload is scaled to minReplicas if there's a problem in metrics delivery (it reports no activity). Or am I mistaken?

That's actually true thing for some scalers, but in those cases the fallback won't apply as those scalers hide the error returning a default value (usually 0), so scaling won't apply either on those cases.
FYI @rickbrouwer , it happens for example with [Prometheus scaler](https://keda.sh/docs/2.16/scalers/prometheus/) and the parameter `ignoreNullValues`, it hides the error just returning 0

---

Thanks for the info @JorTurFer !
What should we do? Should i remove currentReplicas or leave it?

---

I would keep it there personally, for clarity and to not rely on individual scalers behavior, which might change or could be different. With this setting it is straightforward and clean. It doesn't add an issue

---

Let's go ahead with this as it is. Maybe in the future we should default the fallback to this new current replicas to don't have to ways to get the same behavior and that's all